fv_be/app/models/dictionary.py

```diff
@@ -0,0 +1,154 @@
+    """Model for translations associated to each dictionary entry."""
+
+    class TranslationLanguages(models.TextChoices):
+        # Choices for Language
+        ENGLISH = "EN", _("English")
+        FRENCH = "FR", _("French")
+
+    # Fields
+    text = models.CharField(max_length=200)
+    language = models.CharField(
+        max_length=2,
+        choices=TranslationLanguages.choices,
+        default=TranslationLanguages.ENGLISH,
+    )
+    # from fv-word:part_of_speech
+    part_of_speech = models.ForeignKey(
+        PartOfSpeech, on_delete=models.SET_NULL, blank=True, null=True, related_name="translations"
+    )
+    dictionary_entry = models.ForeignKey(
+        "DictionaryEntry", on_delete=models.CASCADE, related_name="translations"
+    )
+
+    def __str__(self):
+        return _("Translation in %(language)s: %(translation)s.") % {
+            "language": self.language,
+            "translation": self.translation,
+        }
+
+
+class AlternateSpelling(BaseSiteContentModel):
+    """Model for alternate spellings associated to each dictionary entry."""
+
+    # from fv:alternate_spelling, fv-word:alternate_spellings, fv-phrase:alternate_spellings
+    text = models.CharField(max_length=200)
+    dictionary_entry = models.ForeignKey(
+        "DictionaryEntry", on_delete=models.CASCADE, related_name="alternate_spellings"
+    )
+
+    def __str__(self):
+        return self.text
+
+
+class Pronunciation(BaseSiteContentModel):
+    """Model for pronunciations associated to each dictionary entry."""
+
+    # from fv-word:pronunciation
+    text = models.CharField(max_length=200)
+    dictionary_entry = models.ForeignKey(
+        "DictionaryEntry", on_delete=models.CASCADE, related_name="pronunciations"
+    )
+
+    def __str__(self):
+        return self.text
+
+
+class DictionaryEntry(BaseControlledSiteContentModel):
+    """Model for dictionary entries"""
+
+    class TypeOfDictionaryEntry(models.TextChoices):
+        # Choices for Type
+        WORD = "WORD", _("Word")
+        PHRASE = "PHRASE", _("Phrase")
+
+    # Fields
+    # from dc:title, relatively more max_length due to phrases
+    # todo: max_length may be modified after doing some analysis on the length of current phrases
+    title = models.CharField(max_length=800)
+    type = models.CharField(
+        max_length=6,
+        choices=TypeOfDictionaryEntry.choices,
+        default=TypeOfDictionaryEntry.WORD,
+    )
+    # todo: Link to site model when available
+    # may be inherited from an abstract base class or mixin later
+    # site = models.ForeignKey(
+    #     'Site',
+    #     on_delete=models.CASCADE
+    # )
+    # from fv-word:categories, fv-phrase:phrase_books
+    categories = models.ForeignKey(
+        Category,
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="dictionary_entries",
+    )
+    # from fv:custom_order
+    custom_order = models.CharField(max_length=200, blank=True)
```

Comment:
I think this will need to have the same length as the title, right? @caforbes @bronson-c

Reply:
It will not be the same length as the title, it could be either shorter or longer.

Reply:
Oh but oops the point is that it should be way closer to the current max value of the title -- probably the same would be safe.